packages/@webex/plugin-cc/src/AgentConfig/AgentConfig.ts

```diff
@@ -0,0 +1,111 @@
+import {IAgentConfig} from './types';
+import AgentConfigService from '../AgentConfigService/AgentConfigService';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from '../AgentConfigService/types';
+import {WebexSDK} from '../types';
+import {DEFAULT_ATTRIBUTES, DEFAULT_PAGE, DEFAULT_PAGE_SIZE} from './constants';
+
+export default class AgentConfig {
+  ciUserId: string;
+  agentProfile: IAgentConfig;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get Agent Profile.
+   * @returns {Promise<AgentProfileResponse>} A promise that eventually resolves to an API response and return configuration of an Agent.
+   * @example
+   * Create an AgentProfile class instance and invoke the getAgentProfile method.
+   * const agentProfile = new AgentProfile('ciUserId', 'webexObject', 'contactCenterApiUrl');
+   * const agentProfileResponse = agentProfile.getAgentProfile();
+   * console.log(JSON.stringify(agentProfileResponse));
+   */
+
+  public async getAgentProfile(): Promise<IAgentConfig> {
+    try {
+      const orgId = await this.webex.credentials.getOrgId();
+
+      const agentConfigService = new AgentConfigService(
+        this.ciUserId,
+        orgId,
+        this.webex,
+        this.wccAPIURL
+      );
+
+      const user: UserResponse = await agentConfigService.getUserUsingCI();
+
+      const agentDesktopProfile: DesktopProfileResponse =
+        await agentConfigService.getDesktopProfileById(user?.agentProfileId);
+      this.agentProfile.loginVoiceOptions = agentDesktopProfile?.loginVoiceOptions;
+
+      const teamListFilter = user?.teamIds;
```

Comment:
None of the below code needs to be executed if agentDesktopProfileId isn't present

Reply:
Code Updated! The attributes are mandatory.